internal/validation/v2alpha1/hosts_test.go

```diff
@@ -0,0 +1,332 @@
+					getHostPtr("host.domain-with-wrong-character-*-inside.com"),
+					getHostPtr("host.example.tld-with-wrong-character-*-inside"),
+					getHostPtr("*-host-starts-with-wrong-character.example.com"),
+					getHostPtr("host.*-domain-starts-with-wrong-character.com"),
+					getHostPtr("host.example.*-tld-starts-with-wrong-character"),
+					getHostPtr("host-ends-with-wrong-character-*.example.com"),
+					getHostPtr("host.domain-ends-with-wrong-character-*.com"),
+					getHostPtr("host.example.tld-ends-with-wrong-character-*"),
+				},
+			},
+		}
+
+		//when
+		problems := validateHosts(".spec", networkingv1beta1.VirtualServiceList{}, apiRule)
+
+		//then
+		Expect(problems).To(HaveLen(9))
+		for i := 0; i < 9; i++ {
+			Expect(problems[i].AttributePath).To(Equal(fmt.Sprintf(".spec.hosts[%d]", i)))
+			Expect(problems[i].Message).To(Equal("Host is not fully qualified domain name"))
+		}
+	})
+
+	It("Should fail if any segment in host name starts or ends with dash", func() {
+		//given
+		apiRule := &v2alpha1.APIRule{
+			Spec: v2alpha1.APIRuleSpec{
+				Hosts: []*v2alpha1.Host{
+					getHostPtr("-host.example.com"),
+					getHostPtr("host-.example.com"),
+					getHostPtr("host.example-.com"),
+					getHostPtr("host.example.-com"),
+				},
+			},
+		}
+
+		//when
+		problems := validateHosts(".spec", networkingv1beta1.VirtualServiceList{}, apiRule)
+
+		//then
+		Expect(problems).To(HaveLen(4))
+		for i := 0; i < 4; i++ {
+			Expect(problems[i].AttributePath).To(Equal(fmt.Sprintf(".spec.hosts[%d]", i)))
+			Expect(problems[i].Message).To(Equal("Host is not fully qualified domain name"))
+		}
+	})
+
+	It("Should fail if any host that is occupied by any Virtual Service exposed by another resource", func() {
+		//given
+		apiRule := &v2alpha1.APIRule{
+			Spec: v2alpha1.APIRuleSpec{
+				Hosts: []*v2alpha1.Host{
+					getHostPtr("host.example.com"),
+					getHostPtr("occupied.example.com"),
+				},
+			},
+		}
+		virtualService1 := &networkingv1beta1.VirtualService{
+			Spec: v1beta1.VirtualService{
+				Hosts: []string{
+					"not-occupied1.example.com",
+					"not-occupied2.example.com"},
+			},
+		}
+		virtualService2 := &networkingv1beta1.VirtualService{
+			Spec: v1beta1.VirtualService{
+				Hosts: []string{
+					"not-occupied3.example.com",
+					"occupied.example.com"},
+			},
+		}
+
+		virtualServiceList := networkingv1beta1.VirtualServiceList{
+			Items: []*networkingv1beta1.VirtualService{
+				virtualService1,
+				virtualService2,
+			},
+		}
+
+		//when
+		problems := validateHosts(".spec", virtualServiceList, apiRule)
+
+		//then
+		Expect(problems).To(HaveLen(1))
+		Expect(problems[0].AttributePath).To(Equal(".spec.hosts[1]"))
+		Expect(problems[0].Message).To(Equal("Host is occupied by another Virtual Service"))
+	})
+
+	It("Should not fail if a host is occupied by the Virtual Service related to the same API Rule", func() {
+		//given
+		apiRule := &v2alpha1.APIRule{
```

Comment:
```suggestion
		apiRule := &v2alpha1.APIRule{
			ObjectMeta: metav1.ObjectMeta{
				Name:      "some-name",
				Namespace: "some-ns",
			},
```

Reply:
just for the same of `getExpectedOwnerLabel` thats building a string out of the metadata